### ISSUE TITLE
Open workspace by typing path

### ISSUE THREAD

**Issue by Adriankhl** [eclipse-theia/theia#8567]
Motivation: I realized that the browser ide cannot recognize network drive in Windows, as mentioned here https://github.com/eclipse-theia/theia/issues/3174, which has been fixed in the electron version by this PR https://github.com/eclipse-theia/theia/issues/3173. However, I cannot see any fix for the browser version.

And actually, after some testing, theia can recognize network drive if there is already a workspace there. I can edit the `recentworkspace.json` to add a workspace to say `"file:///P:/git/test"` such that theia can load that project. Then after restarting theia can recognize my P drive. It would be great if there is some way to input the path other than editing `recentworkspace.json`. Maybe such thing exists but I didn't realize?

At the very least, this can tell people that they can work on network drive by hacking `recentworkspace.json`.

**Comment by vince-fugnitto** [comment]
@Adriankhl thank you for the discussion, where are you expecting to provide the input 
(I assume directly in the `file-dialog`)?

Note that it is possible to start the application with a pre-defined workspace, and should be possible to change workspaces in the browser using the browser's address bar (I have never tried on Windows however with drive letters).

```
http://localhost:3000/#/Users/vincentfugnitto/workspace/theia-apps
```

<div align='center'>

<img width="611" alt="Screen Shot 2020-09-27 at 10 31 41 AM" src="https://user-images.githubusercontent.com/40359487/94367421-ac4c7980-00ac-11eb-9497-0330f3f748fd.png">

</div>

**Comment by Adriankhl** [comment]
> @Adriankhl thank you for the discussion, where are you expecting to provide the input
> (I assume directly in the `file-dialog`)?

I believe so, it is the box opened by clicking `Open Workspace`

> 
> Note that it is possible to start the application with a pre-defined workspace, and should be possible to change workspaces in the browser using the browser's address bar (I have never tried on Windows however with drive letters).
> 
> ```
> http://localhost:3000/#/Users/vincentfugnitto/workspace/theia-apps
> ```

Thanks for the tips, this works well. After going to `http://localhost:3000/#/p:`, I can now view the subdirectory of it to open further workspace belonging to the network drive. I still believe it is better to allow users to input path in  `file-dialog` so the setup is more strict-forward

**Comment by vince-fugnitto** [comment]
> Thanks for the tips, this works well. After going to `http://localhost:3000/#/p:`, I can now view the subdirectory of it to open further workspace belonging to the network drive. I still believe it is better to allow users to input path in `file-dialog` so the setup is more strict-forward

Thank you for the clarification, I believe the issue can be simplified to express that we should enhance the `file-dialog` (both open, open workspace, etc.) in order to allow users the ability to open based on a provided path.

The issue is being tracked as: https://github.com/eclipse-theia/theia/issues/2815

**Comment by vince-fugnitto** [comment]
Following the https://github.com/eclipse-theia/theia/issues/8567#issuecomment-699967854, the issue can be closed as it is a duplicate of #2815.
